CHANGELOG.md

```diff
@@ -4,6 +4,120 @@ All notable changes to Router will be documented in this file.
+
+### use a parking-lot mutex in Context ([Issue #2751](https://github.com/apollographql/router/issues/2751))
+
+The context requires synchronized access to the busy timer, and precedently we used a futures aware mutex for that, but those are susceptible to contention. This replaces that mutex with a parking-lot synchronous mutex that is much faster.
+
+By [@Geal](https://github.com/Geal) in https://github.com/apollographql/router/pull/2885
+
+### Config and schema reloads now use async IO ([Issue #2613](https://github.com/apollographql/router/issues/2613))
+
+If you were using local schema or config then previously the Router was performing blocking IO in an async thread. This could have caused stalls to serving requests and was generally bad practice.
+The Router now uses async IO for all config and schema reloads.
+
+Fixing the above surfaced an issue with the experimental `force_hot_reload` feature introduced for testing. This has also been fixed and renamed to `force_reload`. 
+
+```diff
+experimental_chaos:
+-    force_hot_reload: 1m
++    force_reload: 1m
+```
+
+By [@bryncooke](https://github.com/bryncooke) in https://github.com/apollographql/router/pull/3016
+
+### Improve subgraph co-processor context processing ([Issue #3058](https://github.com/apollographql/router/issues/3058))
+
+Each call to a subgraph co-processor could update the entire request context as a single operation. This is racy and could lead to difficult to predict context modifications depending on the order in which subgraph requests and responses are processed by the router.
+
+This fix modifies the router so that subgraph co-processor context updates are merged within the existing context. This is still racy, but means that subgraphs are only racing to perform updates at the context key level, rather than across the entire context. This is a substantial improvement on the current situation.
+
+Future enhancements will provide a more comprehensive mechanism that will support some form of sequencing or change arbitration across subgraphs.
+
+By [@garypen](https://github.com/garypen) in https://github.com/apollographql/router/pull/3054
+
+## 🛠 Maintenance
+
+### Add a private part to the Context structure ([Issue #2800](https://github.com/apollographql/router/issues/2800))
+
+There's a cost in using the `Context` structure throughout a request's lifecycle, due to the JSON serialization and deserialization, so it should be reserved from inter plugin communication between rhai, coprocessor and Rust. But for internal router usage, we can have a more efficient structure that avoids serialization costs, and does not expose data that should not be modified by plugins.
+
+That structure is based on a map indexed by type id, which means that if some part of the code can see that type, then it can access it in the map.
+
+By [@Geal](https://github.com/Geal) in https://github.com/apollographql/router/pull/2802
```

Comment:
Does this still work?

```suggestion
### Add private component to the `Context` structure ([Issue #2800](https://github.com/apollographql/router/issues/2800))

There's a cost in using the `Context` structure during a request's lifecycle, due to JSON serialization and deserialization incurred when doing inter-plugin communication (e.g., between Rhai/coprocessors and Rust).  For internal router usage, we can use a more efficient structure that avoids serialization costs of our private contextual properties which do not need to be exposed (and shouldn't be modified) by plugins.

The new structure is based on a map indexed by type id, which means that if some part of the code can see that type, then it can access it in the map.

By [@Geal](https://github.com/Geal) in https://github.com/apollographql/router/pull/2802
```